table/block_based_table_reader.cc

```diff
@@ -545,7 +552,7 @@ class HashIndexReader : public IndexReader {
     return index_block_->NewIterator<IndexBlockIter>(
         icomparator_, icomparator_->user_comparator(), iter, kNullStats,
         total_order_seek, index_key_includes_seq_, index_value_is_full_,
-        prefix_index_.get());
+        false /* block_contents_pinned */, prefix_index_.get());
```

Comment:
Can we add an inline comment that for indexes there is no value in having them pinned so we always set it to false.